gnovm/tests/stdlibs/testing/context_testing.gno

```diff
@@ -0,0 +1,138 @@
+package testing
+
+import (
+	"std"
+	"time"
+)
+
+func getContext() (
+	originCaller string,
+	originPkgAddress string,
+	origSendDenoms []string, origSendAmounts []int64,
+	origSpendDenoms []string, origSpendAmounts []int64,
+	chainID string,
+	height int64,
+	timeUnix int64, timeNano int64,
+)
+
+func setContext(
+	originCaller string,
+	originPkgAddress string,
+	currRealmAddr string, currRealmPkgPath string,
+	origSendDenoms []string, origSendAmounts []int64,
+	origSpendDenoms []string, origSpendAmounts []int64,
+	chainID string,
+	height int64,
+	timeUnix int64, timeNano int64,
+)
+
+func GetContext() Context {
+	originCaller, originPkgAddress, origSendDenoms, origSendAmounts, origSpendDenoms, origSpendAmounts, chainID, height, timeUnix, timeNano := getContext()
+	return Context{
+		OriginCaller:     std.Address(originCaller),
+		OriginPkgAddress: std.Address(originPkgAddress),
+		OriginSend:       compactNative(origSendDenoms, origSendAmounts),
+		OriginSpend:      compactNative(origSpendDenoms, origSpendAmounts),
+		ChainID:          chainID,
+		Height:           height,
+		Time:             time.Unix(timeUnix, timeNano),
+	}
+}
+
+func SetContext(ctx Context) {
+	originSendDenom, originSendAmount := expandNative(ctx.OriginSend)
+	originSpendDenom, originSpendAmount := expandNative(ctx.OriginSpend)
+
+	var timestamp, timestampNano int64
+
+	if !ctx.Time.IsZero() {
+		unixNano := ctx.Time.UnixNano()
+		timestamp = unixNano / int64(time.Second)
+		timestampNano = unixNano % int64(time.Second)
+	}
+
+	setContext(
+		ctx.OriginCaller.String(),
+		ctx.OriginPkgAddress.String(),
+		ctx.CurrentRealm.Address().String(), ctx.CurrentRealm.PkgPath(),
+		originSendDenom, originSendAmount,
+		originSpendDenom, originSpendAmount,
+		ctx.ChainID,
+		ctx.Height,
+		timestamp, timestampNano,
+	)
+}
+
+func testIssueCoins(addr string, denom []string, amt []int64)
+
+func SetOriginCaller(origCaller std.Address) {
```

Comment:
```suggestion
func (c *Context) SetOriginCaller(origCaller std.Address) {
```

what about doing this, and also adding a `func (c Context) String()` that we can print?

cc @thehowl

Reply:
I'm not sure about the stringer, I think it can already be printed well with `fmt.Sprintf("%+v", c)`?

Maybe just doing that, not sure.

As for moving to Context, could be good mostly for godoc and "type organization"; in terms of usage, I think most usages would be `testing.GetContext().SetXXX`, so maybe a bit more verbose.